### PR TITLE
Can't create a new wallet with biometry

### PR DESCRIPTION
**Pull request by perunt**
Fixes RNBW-3362

## What changed (plus any additional context for devs)
It applies only to android when using face id/fingerprint.
When creating a new wallet, the goBack function conflicted with the native verification modal; I added a delay so that the modal with the new wallet name had time to close.

## PoW (screenshots / screen recordings)
before: https://streamable.com/lc7c9c
after: https://streamable.com/ppu8zp

## Dev checklist for QA: what to test
Try to create a new wallet with the existing wallet

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Comment by osdnk** [comment]
I have a preference of researching other options, e.g., InteractionManager for this. Also, what does it means "it conflicts". This is a bit vague and I want to deeply understand the reason for this issue. Cause this is just JS modal so why does this stop closing on some native thing popping out?

File: src/components/expanded-state/WalletProfileState.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import analytics from '@segment/analytics-react-native';
+import delay from 'delay';
```
[the remark this answers is not in the document]
Comment (terrysahaidak) [reply src/components/expanded-state/WalletProfileState.js:2]: We actually use this pattern a lot in the app, but the import is wrong for sure:
```suggestion
import { delay } from '@rainbow-me/helpers/utilities';
```

[the remark this answers is not in the document]
Comment (perunt) [reply src/components/expanded-state/WalletProfileState.js:2]: @osdnk okay, I'll keep in it mind

[the remark this answers is not in the document]
Comment (perunt) [reply src/components/expanded-state/WalletProfileState.js:2]: @terrysahaidak thanks, will use that one

**Comment by perunt** [comment]
@osdnk, InteractionManager works as well, so I replaced this part.
I understood that JS freezes when the biometrics authentication window is open, and we try to change the navigation state.  No errors occur. The logger also doesn't show anything useful

**Review by osdnk — APPROVED** [review]
I'm okay with this approach and this is not a big deal. However, I would still appreciate devoting time to understanding why this happens (coz this is weird)

**Review by ibrahimtaveras00 — APPROVED** [review]
Fix looks good, QA Passed 👍🏽

**Comment by terrysahaidak** [comment]
> I'm okay with this approach and this is not a big deal. However, I would still appreciate devoting time to understanding why this happens (coz this is weird) 

When android shows any native view (share view, biometric auth, perhaps alert) it freezes the execution of the app including the js thread, so I'm not sure it's an easy task to figure out what's happening, but maybe it skips the execution of this event loop cycle, not sure, probably even setTimeout with 0 would fix it.

**Review by terrysahaidak — APPROVED** [review]

**Pull request merged**